### DISCUSSION TITLE
How to define polynomial equation constraints in Yalmip Gurobi

### DISCUSSION THREAD

**Discussion by LindsayHNU** [yalmip/YALMIP#1312] [Optimization modelling]
Dear yalmip developers.
Thank you for developing yalmip, it is really excellent in handling optimization problems. I would like to add a polynomial equation constraint to Yalmip Gurobi. The refman manual of Gurobi has the following instructions on page 666.
![image](https://github.com/yalmip/YALMIP/assets/26338611/47224eec-d1a3-4c3e-a518-1c2339995a32)

I am writing a program to output the Gurobi model through Yalmip and then add the following constraint, but this constraint does not work when solving.
![image](https://github.com/yalmip/YALMIP/assets/26338611/5bb0296c-9434-4f3a-9c27-92d3dbe0b143)

`[model,recoverymodel,diagnostic,internalmodel] = export(Cons, Cost_total, sdpsettings('solver','gurobi'));`
`model.genconpoly.xvar=X;`
`model.genconpoly.yvar=Y;`
`model.genconpoly.p=[-0.00003041,0.01901,0.2593,0];`
`model.genconpoly.funcpieces=10;`
`options=sdpsettings('solver','gurobi');`
`sol= optimize(Cons,Cost_total,options);`

I would like to know how to solve this problem and I look forward to hearing from you~

**Comment by johanlofberg** [comment]
That code makes no sense as it appears you want your manipulations of the exported model to magically make their way back to the YALMIP model. Once you export, you've left the YALMIP world.

YALMIP does not support the built-in pwa-modelling of gurobi, but instead has a more general layer of pwa modelling https://yalmip.github.io/command/interp1/ where you for a MILP solver would use the flags sos2 or graph or milp depending on context and wanted model.

**Reply by johanlofberg** [reply]
Best flag is 'lp' as it automatically selects the best possible model based on the data it approximates

**Reply by LindsayHNU** [reply]
OK I'll give it a try, thanks for the guidance (^▽^)
